src/Ledger/Tally.lagda

```diff
@@ -46,32 +47,33 @@ private
            s s' : TallyState
            aid : GovActionID
            role : GovRole
-           kh : KeyHash
+           cred : Credential
            v : Vote
-           md : GovMD
            l : List GovProcedure
            c : Coin
            addr : RwdAddr
            a : GovAction
 \end{code}
 \begin{code}
-addVote : TallyState → GovActionID → GovRole → KeyHash → Vote → TallyState
+addVote : TallyState → GovActionID → GovRole → Credential → Vote → TallyState
 addVote s aid r kh v =
   mapSingleValue (λ s' → record s' { votes = insert (votes s') (r , kh) v }) s aid
 
 addAction : TallyState → Epoch → GovActionID → Coin → RwdAddr → GovAction → TallyState
 addAction s e aid c addr a = insert s aid record
-  { votes = ∅ᵐ ; deposit = c ; returnAddr = addr ; proposedIn = e ; action = a }
+  { votes = ∅ᵐ ; deposit = c ; returnAddr = addr ; expiresIn = e ; action = a }
 
+-- x is the anchor in those two cases, which we don't do anything with
 data _⊢_⇀⦇_,TALLY'⦈_ : TallyEnv → TallyState → GovProcedure → TallyState → Set where
-  TallyVote : let open TallyEnv Γ in
+  TallyVote : ∀ {x} → let open TallyEnv Γ in
     aid ∈ dom (s ˢ)
-    → (kh , role) ∈ roles ˢ
     ────────────────────────────────
-    Γ ⊢ s ⇀⦇ vote aid role kh v md ,TALLY'⦈ addVote s aid role kh v
+    Γ ⊢ s ⇀⦇ vote aid role cred v x ,TALLY'⦈ addVote s aid role cred v
 
-  TallyPropose : let open TallyEnv Γ in
-    Γ ⊢ s ⇀⦇ propose c addr a md ,TALLY'⦈ addAction s epoch (txid , length l) c addr a
+  TallyPropose : ∀ {x} → let open TallyEnv Γ; open PParams pparams using (govExpiration; govDeposit) in
+    c ≡ govDeposit
```

Comment:
This also needs a check for `expiresIn`

Reply:
`expiresIn` is set to `current epoch + govExpiration` in `addAction`. So we don't need to check it, because the transaction doesn't get a say in what it is.

Reply:
Ah, I see